lib/testutils/httpmultibin/httpmultibin.go

```diff
@@ -253,15 +252,29 @@ func (*GRPCStub) HalfDuplexCall(grpctest.TestService_HalfDuplexCallServer) error
 }
 
 // NewHTTPMultiBin returns a fully configured and running HTTPMultiBin
-func NewHTTPMultiBin(t testing.TB) *HTTPMultiBin {
+//nolint:funlen
+func NewHTTPMultiBin(t testing.TB, opts ...MultiBinOptWrap) *HTTPMultiBin {
+	// TODO instead of having opts maybe make the whole mutlibin more composable, as there is no point for most of the
+	// servers inthe majority of the tests
+	var customConnContext func(ctx context.Context, c net.Conn) context.Context
+	for _, opt := range opts {
+		switch opt.Key {
+		case NullOpt:
+			t.Fatalf("unsupported multibin option %d, the NullOpt should not be used", opt.Key)
```

Comment:
What's the purpose of defining/checking for it then?

Reply:
Literally a linter told me I don't check all of them and it is nice to have a more specific message for why it failed. On the other hand, maybe I should add that this will happen also if `Key` wasn't set as it will default to 0 :thinking:

Reply:
OK, but I mean why even define `NullOpt` at all then? It's not used anywhere, and you could start `ConnContextFuncOpt` at `iota + 1` if you don't want to use `0`. Though it's a minor thing, I don't mind it much, just wondering.

Reply:
hm .... I guess given that this is not user-facing code it might be easier if I just drop it :thinking: in general I prefer to have it for user-facing enums so we can see when it isn't set and provide an error.